gr-blocks/lib/selector_impl.cc

```diff
@@ -152,10 +159,32 @@ int selector_impl::general_work(int noutput_items,
 
     gr::thread::scoped_lock l(d_mutex);
     if (d_enabled) {
+
+        // propagate tags
+        std::vector<tag_t> tags;
+        get_tags_in_window(tags, 0, 0, noutput_items);
+        for (auto tag : tags) {
+            tag.offset -= d_dropped_samples[d_output_index];
```

Comment:
If I am not mistaken, `tag.offset = nitems_written(d_output_index)` could be used instead of tracking the number of dropped samples.